moztelemetry/dataset.py

```diff
@@ -169,6 +183,8 @@ def where(self, **kwargs):
                 raise Exception('The dimension {} doesn\'t exist'.format(dimension))
             if isfunction(condition) or isinstance(condition, functools.partial):
                 clauses[dimension] = condition
+            elif isinstance(condition, basestring):
```

Comment:
The value that this condition is checked against is a string, so we can either raise an exception if this is not the case or convert condition to a string. I think the latter solution would make more sense but I don't have a strong opinion about it.